### PR TITLE
Add account CCPA enabled and per-request-type enabled flags

### PR DESCRIPTION
**Pull request by snahornyi**

### CODE REVIEW

File: src/main/java/org/prebid/server/auction/PrivacyEnforcementService.java [hunk 1]
```diff
@@ -199,21 +201,55 @@ private static RequestLogInfo requestLogInfo(MetricName requestType, BidRequest
...
 
         return shouldEnforceCcpa && ccpa.isEnforced();
     }
 
+    private boolean isCcpaEnforced(Ccpa ccpa, Account account, MetricName requestType) {
+        final boolean shouldEnforceCcpa = isCCPAEnabled(account, requestType);
+        return shouldEnforceCcpa && ccpa.isEnforced();
+    }
+
+    private Boolean isCCPAEnabled(Account account) {
```
Comment (DGarbar) [inline src/main/java/org/prebid/server/auction/PrivacyEnforcementService.java:225]: Use camelCase

File: src/main/java/org/prebid/server/auction/PrivacyEnforcementService.java [hunk 2]
```diff
@@ -199,21 +201,55 @@ private static RequestLogInfo requestLogInfo(MetricName requestType, BidRequest
...
+        final AccountCcpaConfig accountCcpaConfig = account.getCcpa();
+
+        final Boolean accountCCPAEnabled =
+                accountCcpaConfig != null ? accountCcpaConfig.getEnabled() : null;
```
Comment (DGarbar) [inline src/main/java/org/prebid/server/auction/PrivacyEnforcementService.java:229]: redundant /n

File: src/main/java/org/prebid/server/auction/PrivacyEnforcementService.java [hunk 3]
```diff
@@ -199,21 +201,55 @@ private static RequestLogInfo requestLogInfo(MetricName requestType, BidRequest
...
     }
 
+    private boolean isCcpaEnforced(Ccpa ccpa, Account account, MetricName requestType) {
+        final boolean shouldEnforceCcpa = isCCPAEnabled(account, requestType);
+        return shouldEnforceCcpa && ccpa.isEnforced();
+    }
+
+    private Boolean isCCPAEnabled(Account account) {
+        final AccountCcpaConfig accountCcpaConfig = account.getCcpa();
+
```
Comment (DGarbar) [inline src/main/java/org/prebid/server/auction/PrivacyEnforcementService.java:227]: redundant /n

File: src/main/java/org/prebid/server/auction/PrivacyEnforcementService.java [hunk 4]
```diff
@@ -199,21 +201,55 @@ private static RequestLogInfo requestLogInfo(MetricName requestType, BidRequest
...
+        final Boolean accountCCPAEnabled =
+                accountCcpaConfig != null ? accountCcpaConfig.getEnabled() : null;
+
+        return ObjectUtils.firstNonNull(accountCCPAEnabled, account.getEnforceCcpa(), ccpaEnforce);
+    }
+
+    private boolean isCCPAEnabled(Account account, MetricName requestType) {
+        final AccountCcpaConfig accountCcpaConfig = account.getCcpa();
+
+        final Boolean accountCCPAEnabled =
```
Comment (DGarbar) [inline src/main/java/org/prebid/server/auction/PrivacyEnforcementService.java:237]: redundant /n

File: src/main/java/org/prebid/server/auction/PrivacyEnforcementService.java [hunk 5]
```diff
@@ -199,21 +201,55 @@ private static RequestLogInfo requestLogInfo(MetricName requestType, BidRequest
...
+                accountCcpaConfig != null ? accountCcpaConfig.getEnabled() : null;
+        if (requestType == null) {
+            return ObjectUtils.firstNonNull(accountCCPAEnabled, ccpaEnforce);
```
Comment (DGarbar) [inline src/main/java/org/prebid/server/auction/PrivacyEnforcementService.java:240]: Missed  `eturn ObjectUtils.firstNonNull(accountCCPAEnabled, account.getEnforceCcpa(), ccpaEnforce);`

**Review by DGarbar — CHANGES_REQUESTED** [review]
@rpanchyk  I Think this feature is also connected with `Account.enforceCcpa`. 
Probably it is better to migrate this to `Account.ccpa.enabled`. 
So we need to change DB schema as well. But we can merge this ticket as it is, bc current workflow is no changed

**Comment by DGarbar** [comment]
Current code is OK. But I think we need to change JdbcApplicationSettings and DB. Lets wait for Ruslan

**Review by rpanchyk — APPROVED** [review]

**Review by rpanchyk — APPROVED** [review]

**Pull request merged**